api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -243,13 +291,20 @@ private void processIdleRuns(final List<PipelineRun> runs) {
                 double cpuUsageRate = metric / MILLIS / type.getVCPU();
                 if (Precision.compareTo(cpuUsageRate, idleCpuLevel, ONE_THOUSANDTH) < 0) {
                     processIdleRun(run, actionTimeout, action, runsToNotify, runsToUpdate, cpuUsageRate);
+                    if (run.getTags().put(UTILIZATION_LEVEL_LOW, TRUE_VALUE_STRING) == null) {
+                        runsToUpdateTags.add(run);
+                    }
                 } else if (run.getLastIdleNotificationTime() != null) { // No action is longer needed, clear timeout
                     run.setLastIdleNotificationTime(null);
+                    final String tagToBeRemoved = run.getTags().remove(UTILIZATION_LEVEL_LOW);
+                    if (tagToBeRemoved != null
```

Comment:
take a look at method remove(key, value). I believe it could be suitable here as well